io.openems.backend.metadata.odoo/src/io/openems/backend/metadata/odoo/MetadataOdoo.java

```diff
@@ -200,7 +200,8 @@ public Optional<User> getUser(String userId) {
 
 	@Override
 	public Collection<Edge> getAllOfflineEdges() {
-		return this.edgeCache.getAllEdges().stream().filter(Edge::isOffline).toList();
+		return this.edgeCache.stream().filter(Edge::isOffline) //
+				.map(e -> (Edge) e).toList();
```

Comment:
Best practice in OpenEMS for streams is, to have every stream operation in a separate file, i.e. `filter()` should be in the next line.